### ISSUE TITLE
tcp_listen does not seem to error if port is "taken"

### ISSUE THREAD

**Issue by YellowAfterlife** [Medo42/Faucet-Networking-Extension#4]
As such, you can attempt to listen on a port that is taken by other application (or even another acceptor from the same application), and no error will be raised, but (unsurprisingly) this will not actually bind the acceptor to a port and incoming connections will go to the existing acceptor while the new acceptor pretends that everything is alright.

Sample:
```gml
s1 = tcp_listen(5394)
s2 = tcp_listen(5394)
show_debug_message("s1: id" + string(s1) + "; hasError: " + string(socket_has_error(s1)) + "; " + socket_error(s1))
show_debug_message("s2: id" + string(s2) + "; hasError: " + string(socket_has_error(s2)) + "; " + socket_error(s2))
```
outputs:
```
s1: id1; hasError: 0; IPv4: ; IPv6:
s2: id2; hasError: 0; IPv4: ; IPv6: 
```
Or is this something of that "Please note that acceptors will only flag an error if both
IPv4 and IPv6 connections can't be accepted anymore."? (although I would imagine that both acceptors should try to occupy the same or both protocols)

Isn't a large problem but it's generally good to be able to tell if the game definitely can't be hosted on the specified port, for example.

**Comment by Medo42** [comment]
I'll have to look into this a bit more in-depth before I can tell you more (probably next weekend), but I agree that this behaviour is unfortunate. It might have been a mistake to try "transparently" supporting IPv6 by automatically binding sockets for both protocols - this was inspired by Vista's Dual Stack sockets, but since Faucet Net supports Windows XP as well I had to grow my own solution involving two separate sockets. Looking back at that, there should at least be functions to figure out whether a given acceptor is active and ok for each protocol individually.
The lack of error when using an existing address is probably due to the use of `SO_REUSEADDR`, though I'm a bit surprised that you don't get any error when binding to a port already in use by a different program - maybe it does successfully create an IPv6 listening socket and thinks that's good enough. If that's the case, functions to check if a specific protocol is working OK would solve that particular aspect.
It's also possible that the other program was listening on a particular adapter IP, while Faucet Net is listening for connections from anywhere (0.0.0.0). I'm not sure how that would work out, but I could imagine that nobody sees an error and the socket bound to the specific IP gets all the connections.
I'll check why I thought SO_REUSEADDR was a good idea, and how Faucet Net can ensure you notice something is wrong in these situations.

**Comment by Medo42** [comment]
I looked over the code and don't really understand why I added `SO_REUSEADDR`. After removing it, everything still seems to work as intended on Win7, and you get a proper error when the port is already in use by either this or a different process (but still only if both protocols are unavailable).

I still have to test on XP, but if that also works I'll add extra functions to check whether an acceptor is listening for IPv4 / IPv6 connections and make that version 1.7.0.
